modules/core/src/main/java/org/apache/ignite/internal/processors/bulkload/BulkLoadProcessor.java

```diff
@@ -91,15 +103,17 @@ public BulkLoadCacheWriter outputStreamer() {
      * @throws IgniteIllegalStateException when called after {@link #close()}.
      */
     public void processBatch(byte[] batchData, boolean isLastBatch) throws IgniteCheckedException {
-        if (isClosed)
-            throw new IgniteIllegalStateException("Attempt to process a batch on a closed BulkLoadProcessor");
+        try (MTC.TraceSurroundings ignored = MTC.supportContinual(qrySpan)) {
```

Comment:
Seems that it'll be useful to add some meta to span with an amount of processed batchData, exceptions, etc. I believe that your "tag-counter-listeners" suites great here.

Reply:
Honestly, I'd like to add statistic gathering in the next step after its discussion. It seems that ability to gather statistics should be reusable and not related to span mechanics as proposed earlier.